skimage/graph/_mcp.pyx

```diff
@@ -415,6 +428,9 @@ cdef class MCP:
             end-position will be found; otherwise the algorithm will stop when
             a a path is found to any end-position. (If no `ends` were
             specified, then this parameter has no effect.)
+        max_step_cost : float, optional
+            Cost limit for each step between points. Points whose costs is
```

Comment:
I could think of how to formulate this well, but a point cannot really have a cost, so how do we say: "a barrier is effectively in place in between any two points for which the stepping cost is higher than `max_step_cost`"